watchdog-config/extra-sql-queries.yml

```diff
@@ -75,3 +75,400 @@ users_with_insificient_disk:
+    - partner_name:
+        usage: "LABEL"
+    - last_sync:
+        usage: "LABEL"
+    - days_since_last_sync:
+        usage: "GAUGE"
+    - chw_username:
+        usage: "LABEL"
+    - chw_id:
+        usage: "LABEL"
+    - cht_instance:
+        usage: "LABEL"
+
+gateway_health:
+  query: |
+    SELECT
+        gw.partner_name,
+        reported::varchar(255),
+        due,
+        delivered,
+        scheduled,
+        gateway_state,
+        last_delivered,
+        replace(monitoring_urls.url, 'https://', '') as cht_instance
+    FROM app_monitoring_gateway_health AS gw
+    LEFT JOIN
+        monitoring_urls ON gw.partner_name=monitoring_urls.partner_name
+  metrics:
+    - partner_name:
+        usage: "LABEL"
+    - reported:
+        usage: "LABEL"
+    - due:
+        usage: "GAUGE"
+    - delivered:
+        usage: "LABEL"
+    - scheduled:
+        usage: "LABEL"
+    - gateway_state:
+        usage: "LABEL"
+    - last_delivered:
+        usage: "LABEL"
+    - cht_instance:
+        usage: "LABEL"
+
+invalid_docs:
+  query: |
+    SELECT
+        docs.partner_name,
+        day::varchar(255) AS reported,
+        invalid_doc_count,
+        replace(monitoring_urls.url, 'https://', '') as cht_instance
+    FROM app_monitoring_invalid_docs AS docs
+    LEFT JOIN
+        monitoring_urls ON docs.partner_name=monitoring_urls.partner_name
+  metrics:
+    - partner_name:
+        usage: "LABEL"
+    - reported:
+        usage: "LABEL"
+    - invalid_doc_count:
+        usage: "GAUGE"
+    - cht_instance:
+        usage: "LABEL"
+
+invalid_doc_errors:
+  query: |
+    SELECT
+        docs.partner_name,
+        day::varchar(255) AS reported,
+        code,
+        count,
+        replace(monitoring_urls.url, 'https://', '') as cht_instance
+    FROM app_monitoring_invalid_doc_errors AS docs
+    LEFT JOIN
+        monitoring_urls ON docs.partner_name=monitoring_urls.partner_name
+  metrics:
+    - partner_name:
+        usage: "LABEL"
+    - reported:
+        usage: "LABEL"
+    - code:
+        usage: "LABEL"
+    - count:
+        usage: "GAUGE"
+    - cht_instance:
+        usage: "LABEL"
+
+tasks_by_state:
+  query: |
+    SELECT
+        tasks.partner_name,
+        day::varchar(255) AS reported,
+        task,
+        draft,
+        ready,
+        cancelled,
+        completed,
+        failed,
+        replace(monitoring_urls.url, 'https://', '') as cht_instance
+    FROM app_monitoring_tasks_by_state AS tasks
+    LEFT JOIN
+        monitoring_urls ON tasks.partner_name=monitoring_urls.partner_name
+  metrics:
+    - partner_name:
+        usage: "LABEL"
+    - reported:
+        usage: "LABEL"
+    - task:
+        usage: "LABEL"
+    - draft:
+        usage: "GAUGE"
+    - ready:
+        usage: "GAUGE"
+    - cancelled:
+        usage: "GAUGE"
+    - completed:
+        usage: "GAUGE"
+    - failed:
+        usage: "GAUGE"
+    - cht_instance:
+        usage: "LABEL"
+
+tasks_by_user:
+  query: |
+    SELECT
```

Comment:
this query returns 32k results - any way to trim it down?

Reply:
Limited to last 90 days

Reply:
please revisit this - it looks like it didn't get added to [this query when I check the most recent version](https://github.com/medic/cht-app-monitoring-data-ingest/blob/34e28d277794ec2607bd0d4cc6178b9ad5a2f5a6/watchdog-config/extra-sql-queries.yml#L405).

Reply:
Updated - I had limited tasks_by_state to last 90 days and realized tasks_by_user wouldn't have a reduction in row count by limiting time window. However, it does makes sense to only view tasks in the last 90 days per user.

Reply:
while you did add an extra where clause `AND day >= NOW() - INTERVAL '90 days'` , there is no `day` column so the query fails.

Maybe we [can try to do a](https://stackoverflow.com/questions/2318398/how-can-i-limit-number-of-results-by-a-specific-column-in-postgresql) `PARTITION BY`?  I just found that searching around, I don't have a good option here :shrug: